packages/core/src/video/metrics/utils.ts

```diff
@@ -8,22 +7,68 @@ const RECORDING_URL_PART_VALUE = 'recordings';
 // Example url the playback id needs to be found in
 // https://livepeercdn.com/hls/<playback-id>/index.m3u8
 // https://livepeercdn.com/recordings/<playback-id>/index.m3u8
-function playbackIdFromPlaybackUrl(url: string): string | undefined {
-  const parts = url.split('/');
-  const playlistPartIndex = parts.indexOf(PLAYLIST_NAME);
-  const assetPartIndex = parts.indexOf(ASSET_URL_PART_VALUE);
-  const recordingPartIndex = parts.indexOf(RECORDING_URL_PART_VALUE);
-
-  // Check if the url is valid
-  return (assetPartIndex !== -1 || recordingPartIndex !== -1) &&
-    playlistPartIndex !== -1
-    ? parts[playlistPartIndex - 1]
-    : undefined;
-}
-
-export function createMetricsReportingUrl(src: string): string | undefined {
-  const videoName = playbackIdFromPlaybackUrl(src);
-  return videoName
-    ? `${METRICS_REPORTING_BASE_URL}/json_video+${videoName}.js`
-    : undefined;
-}
+export const getMetricsReportingUrl = async (
+  src: string,
+): Promise<string | null> => {
+  try {
+    const parsedUrl = new URL(src);
+
+    const parts = parsedUrl.pathname.split('/');
+    const playlistPartIndex = parts.indexOf(PLAYLIST_NAME);
+
+    const includesAssetUrl = parts.includes(ASSET_URL_PART_VALUE);
+    const includesRecording = parts.includes(RECORDING_URL_PART_VALUE);
+
+    // Check if the url is valid
+    const playbackId =
+      (includesAssetUrl || includesRecording) && playlistPartIndex !== -1
+        ? parts?.[playlistPartIndex - 1] ?? null
+        : null;
+
+    const splitHost = parsedUrl.host.split('.');
+    const tld = (splitHost?.[splitHost?.length - 1] ?? null) as
+      | 'com'
+      | 'studio'
+      | 'fun'
+      | 'monster'
+      | null;
+
+    // map to known TLDs, with .com => .studio
+    const tldMapped =
+      tld === 'com'
+        ? 'studio'
+        : tld === 'studio'
+        ? 'studio'
+        : tld === 'fun'
+        ? 'fun'
+        : tld === 'monster'
+        ? 'monster'
+        : null;
+
+    // if not a known TLD, then do not return a URL
+    if (playbackId && tldMapped) {
```

Comment:
**nit:** WDYT of inverting the `if` for less indentation?
```suggestion
    // if not a known TLD, then do not return a URL
    if (!playbackId || !tldMapped) {
      return null;
    }
    // proceed
```

Reply:
I actually prefer the style of indentation over early return - just my style though

Reply:
Gotcha! Yeah just follow your style then 😄